uplink/storage/streams/store.go

```diff
@@ -196,10 +193,74 @@ func (s *streamStore) upload(ctx context.Context, path Path, pathCipher storj.Ci
 		if err != nil {
 			return Meta{}, currentSegment, streamID, err
 		}
-		var transformedReader io.Reader
+
+		segmentEncryption := storj.SegmentEncryption{}
+		if s.cipher != storj.EncNull {
+			segmentEncryption = storj.SegmentEncryption{
+				EncryptedKey:      encryptedKey,
+				EncryptedKeyNonce: keyNonce,
+			}
+		}
+
 		if isRemote {
+			encrypter, err := encryption.NewEncrypter(s.cipher, &contentKey, &contentNonce, s.encBlockSize)
+			if err != nil {
+				return Meta{}, currentSegment, streamID, err
+			}
+
 			paddedReader := eestream.PadReader(ioutil.NopCloser(peekReader), encrypter.InBlockSize())
-			transformedReader = encryption.TransformReader(paddedReader, encrypter, 0)
+			transformedReader := encryption.TransformReader(paddedReader, encrypter, 0)
+
+			beginSegment := &metainfo.BeginSegmentParams{
+				MaxOrderLimit: s.maxEncryptedSegmentSize,
+				Position: storj.SegmentPosition{
+					Index: int32(currentSegment),
+				},
+			}
+
+			var responses []metainfo.BatchResponse
+			if currentSegment == 0 {
+				responses, err = s.metainfo.Batch(ctx, []metainfo.BatchItem{
+					beginObjectReq,
+					beginSegment,
+				}...)
```

Comment:
Why are we using this complex notation instead of just:
```go
responses, err = s.metainfo.Batch(ctx, beginObjectReq, beginSegment)
```
I noticed this on other places too.

Reply:
good point, I don't know why I did it like that :)

Reply:
probably dragged by the refactoring goal. I missed too.